plugins/native/src/main/java/org/qbicc/plugin/native_/NativeInfo.java

```diff
@@ -303,50 +309,111 @@ private boolean isArgConst(DefinedTypeDefinition definedType) {
         return false;
     }
 
-    public ValueType getTypeOfFunctionalInterface(final DefinedTypeDefinition definedType) {
-        MethodElement method = getFunctionalInterfaceMethod(definedType);
-        if (method == null) {
+    public class FunctionalInterfaceData {
```

Comment:
Should be `static` (and `final`, as a stylistic consideration) and doesn't need to be `public` (it's inside of a non-`public` class).